sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -266,15 +305,32 @@ case class AlterViewAsCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
-    if (session.sessionState.catalog.alterTempViewDefinition(name, analyzedPlan)) {
-      // a local/global temp view has been altered, we are done.
+    if (session.sessionState.catalog.isTemporaryTable(name)) {
+      alterTemporaryView(session, analyzedPlan)
     } else {
       alterPermanentView(session, analyzedPlan)
     }
-
     Seq.empty[Row]
   }
 
+  private def alterTemporaryView(session: SparkSession, analyzedPlan: LogicalPlan): Unit = {
+    val tableDefinition = if (conf.storeAnalyzedPlanForView) {
+      analyzedPlan
+    } else {
+      val properties = generateViewProperties(
+        Map.empty, session, analyzedPlan, analyzedPlan.schema.fieldNames)
+
+      TemporaryViewRelation(CatalogTable(
```

Comment:
can we call `prepareTemporaryView`?

Reply:
we can if we move `prepareTemporaryView` to `ViewHelper`, but it's a little bit weird to keep `prepareTemporaryView` and `prepareTable` in 2 different class/object

Reply:
I think it's OK, `ViewHelper` should contain view related utils.